vscode-lean4/src/utils/converters.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * For LSP communication, we need a way to translate between LSP types and corresponding VSCode types.
+ * By default this translation is provided as a bunch of methods on a `LanguageClient`, but this is
+ * awkward to use in multi-client workspaces wherein we need to look up specific clients. In fact the
+ * conversions are *not* stateful, so having them depend on the client is unnecessary. Instead, we
+ * provide global converters here.
+ *
+ * Some of the conversions are patched to support extended Lean-specific structures.
+ *
+ * @module
+ */
+
+import { createConverter as createP2CConverter } from 'vscode-languageclient/lib/common/protocolConverter';
+import { createConverter as createC2PConverter } from 'vscode-languageclient/lib/common/codeConverter';
+import * as ls from 'vscode-languageserver-protocol'
+import * as code from 'vscode'
+import { Code2ProtocolConverter, Protocol2CodeConverter } from 'vscode-languageclient';
+
+interface Lean4Diagnostic extends ls.Diagnostic {
+    fullRange: ls.Range;
+}
+
+export const p2cConverter = createP2CConverter(undefined, true, true)
+export const c2pConverter = createC2PConverter(undefined)
```

Comment:
I'm not sure about this, the default for `uriConverters` in `LanguageClientOptions` is:
```
uriConverters?: {
        code2Protocol: c2p.URIConverter;
        protocol2Code: p2c.URIConverter;
    };
```
which means this initialization code in BaseLanguageClient is passing `c2p.URIConverter` instead of `undefined`, no?
```
this._c2p = c2p.createConverter(clientOptions.uriConverters ? clientOptions.uriConverters.code2Protocol : undefined);
this._p2c = p2c.createConverter(
	clientOptions.uriConverters ? clientOptions.uriConverters.protocol2Code : undefined,
	this._clientOptions.markdown.isTrusted,
	this._clientOptions.markdown.supportHtml);
```

Reply:
Ah, you are right, I do see undefined at runtime:
![image](https://user-images.githubusercontent.com/18707114/170128478-e0928b12-9f94-4e52-869c-c62629135a2e.png)